src/neural-network.js

```diff
@@ -35,12 +35,27 @@ export default class NeuralNetwork {
     };
   }
 
+  static get trainingValidations() {
+    return {
+      iterations: (val) => { return typeof val === 'number' && val > 0; },
+      errorThresh: (val) => { return typeof val === 'number' && val > 0 && val < 1; },
+      log: (val) => { return typeof val === 'function' || typeof val === 'boolean'; },
+      logPeriod: (val) => { return typeof val === 'number' && val > 0; },
+      learningRate: (val) => { return typeof val === 'number' && val > 0 && val < 1; },
+      momentum: (val) => { return typeof val === 'number' && val > 0 && val < 1; },
+      callback: (val) => { return typeof val === 'function' || val === null },
+      callbackPeriod: (val) => { return typeof val === 'number' && val > 0; },
+      timeout: (val) => { return typeof val === 'number' && val > 0 }
+    }
+  }
+
   constructor(options = {}) {
     Object.assign(this, this.constructor.defaults, options);
     this.hiddenSizes = options.hiddenLayers;
     this.trainOpts = {};
     this._updateTrainingOptions(Object.assign({}, this.constructor.trainDefaults, options));
 
+    this._isStrict = true;
```

Comment:
If we are going to have multiple validation options, this should be contained within an object that contains the options and this as well. If we are not going to have multiple validation options, this should be named to something more descriptive, e.g. `this._strictTrainingOptionValidation` or something

Reply:
totally agree the name didn't fit.  I ended up changing it before I saw this comment.